src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs

```diff
@@ -165,16 +141,25 @@ static int IndexOfWhiteSpace(string s)
 
             foreach (var group in lockFile.ProjectFileDependencyGroups)
             {
-                foreach (string dependency in group.Dependencies)
+                if (GetFrameworksMatching(group.FrameworkName, frameworkAlias))
                 {
-                    string packageName = GetPackageNameFromDependency(dependency);
-                    set.Add(packageName);
+                    foreach (string dependency in group.Dependencies)
+                    {
+                        string packageName = GetPackageNameFromDependency(dependency);
+                        set.Add(packageName);
+                    }
                 }
             }
 
             return set;
         }
 
+        private static bool GetFrameworksMatching(string framework1, string framework2)
```

Comment:
I would go with something like this:

```suggestion
        private static string GetFrameworkAliasForDependencyGroup(ProjectFileDependencyGroup group)
```

And then at the call site check whether it matches the frameworkAlias (or one or both is null or empty).